### DISCUSSION TITLE
recent update to 5.2.3 causes ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==

### DISCUSSION THREAD

**Discussion by GabLeRoux** [celery/celery#7202] [General]
Hi, I've created an issue on `astroid` which seems to be introduced by Celery 5.2.3 and cannot be reproduced with 5.2.2:  
https://github.com/PyCQA/astroid/issues/1329

I think the issue should be created in celery project directly, but the template asks a lot of questions. It said there should be a discussion first in the checklist so here's the discussion. All details are in linked issue above :pray:

**Comment by auvipy** [comment]
which tool you are using? pip-tools?

**Reply by GabLeRoux** [reply]
Yes, `pip-tools`. Linked issue description has exact steps to reproduce. Issue was closed, most likely because it's unrelated to 
`astroid` itself [as explained here](https://github.com/PyCQA/astroid/issues/1329#issuecomment-1005230126).

It can be reproduced by using `pip-tools`'s `pip-compile --generate-hashes` and a `requirements.in` that looks like this:
```
celery==5.2.3
pylint==2.11.2
```

I've created the following repository which reproduces the issue in a github-action pipeline:  
https://github.com/GabLeRoux/pylint-celery-setuptools-issue

**Reply by auvipy** [reply]
we don' provide pip-tools compliant release yet as it's not official. had to restrict setuptools version for this release.

**Reply by GabLeRoux** [reply]
A simple workaround for this to work with `pip-tools` is to remove `--generate-hashes` when using `pip-compile` or add `--allow-unsafe`. 

When using `--generate-hashes --allow-unsafe`, the generated `requirements.txt` will have the following diff:

```diff
- # WARNING: The following packages were not pinned, but pip requires them to be
- # pinned when the requirements file includes hashes. Consider using the --allow-unsafe flag.
- # setuptools
+ # The following packages are considered to be unsafe in a requirements file:
+ setuptools==59.6.0 \
+     --hash=sha256:22c7348c6d2976a52632c67f7ab0cdf40147db7789f9aed18734643fe9cf3373 \
+     --hash=sha256:4ce92f1e1f8f01233ee9952c04f6b81d1e02939d6e1b488428154974a4d0783e
+     # via
+     #   celery
```

That's good enough, so no worries 👍

**Reply by DmytroLitvinov** [reply]
Hi @auvipy ,

>  had to restrict setuptools version for this release.

Does it mean that future release 5.2.4 will no have anymore strict restriction for setuptools?

**Reply by auvipy** [reply]
yes

**Reply by ThiefMaster** [reply]
Do you have an ETA for 5.2.4? I was about to make a new release of my application that depends on Celery, but I'd really prefer to not have this setuptools pin included there...

**Reply by auvipy** [reply]
may be by the end of this month or 25th jan at best

**Reply by ThiefMaster** [reply]
OK thanks, then I'll downgrade to 5.2.2 for now
